tutorial/beginner/t11_debugging.ipynb

```diff
@@ -0,0 +1,854 @@
+    "* Running the training loop"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6f9e04a8",
+   "metadata": {},
+   "source": [
+    "#### Using Pipeline.transform"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "c291ba7d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class AddNoiseDebug(NumpyOp):\n",
+    "    def __init__(self, inputs, outputs, mode = None):\n",
+    "        super().__init__(inputs, outputs, mode)\n",
+    "\n",
+    "    def forward(self, data, state):\n",
+    "        noise = np.random.normal(0, 1, size=data.shape)\n",
+    "        print('Noise shape ', noise.shape)\n",
+    "        print('data shape ', data.shape) # add print statement to check the data and noise\n",
+    "        data = data + noise\n",
+    "        return data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "cdd6ccec",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "debug_pipeline = fe.Pipeline(train_data=train_data,\n",
+    "                       eval_data=eval_data,\n",
+    "                       test_data=test_data,\n",
+    "                       batch_size=3,\n",
+    "                       ops=[AddNoiseDebug(inputs='x', outputs='x_out'),\n",
+    "                            Rotate(image_in=\"x_out\", image_out=\"x_out\", limit=60)], \n",
+    "                             num_process=0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "889fefdb",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Noise shape  (28, 28)\n",
+      "data shape  (28, 28)\n"
+     ]
+    }
+   ],
+   "source": [
+    "results = debug_pipeline.transform(train_data[0], mode='train')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c90c3da0",
+   "metadata": {},
+   "source": [
+    "#### Running the Training Loop"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "e8c868b2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from fastestimator.op.tensorop.loss import CrossEntropy\n",
+    "from fastestimator.op.tensorop.model import ModelOp, UpdateOp"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "b60b14eb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "network = fe.Network(ops=[ModelOp(model=model, inputs=\"x_out\", outputs=\"y_pred\"),\n",
+    "                          CrossEntropy(inputs=(\"y_pred\", \"y\"), outputs=\"ce\", mode=\"!infer\"),\n",
+    "                          UpdateOp(model=model, loss_name=\"ce\", mode=\"train\")])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9ec29521",
+   "metadata": {},
+   "source": [
+    "NOTE: The training logs will print the extra debug messages if warmup is not set to False in `Estimator.fit`. It will perform a test run on both `Pipeline` and `Network` to make sure that training do not fail in later stage. "
```

Comment:
```suggestion
    "NOTE: The training logs will print extra debug messages if warmup is not set to False in `Estimator.fit`. When warmup is enabled it will perform a test run on both `Pipeline` and `Network` to make sure that training will not fail in later stages. "
```